src/java/org/apache/cassandra/schema/TableMetadata.java

```diff
@@ -461,57 +461,109 @@ public static boolean isSuperColumnMapColumnName(ByteBuffer columnName)
         return !columnName.hasRemaining();
     }
 
+    public void validateTableStructureCompatibility(TableMetadata previous)
+    {
+        if (isIndex())
+            return;
+
+        validateTableName(previous);
```

Comment:
Since the name of the method refers to the structure, maybe it's better to leave the name of the table out of these checks? If we need to validate both, the caller can invoke both `validateTableName` and  `validateTableStructureCompatibility`.

Reply:
The existing implementation of `validateCompatibility()` was already comparing the names of the keyspace and table, so it seemed right to keep the table name comparison in my modified version (while I had to take out the keyspace comparison due to the tenant prefix). It is a check that needs to be done, so ideally I'd leave it in this method.

I take your point that the name of my modified method is not accurate though. Would `validateTableNameAndStructureCompatibility()` be better, or is it too verbose?